pkg/linters/funcguard/forbiddenusage.go

```diff
@@ -0,0 +1,113 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2024-present Datadog, Inc.
+
+// Package funcguard provides is a linter to detect unwanted/deprecated function calls in the code base.
+package funcguard
+
+import (
+	"fmt"
+	"go/ast"
+	"strings"
+
+	"github.com/golangci/plugin-module-register/register"
+	"github.com/mitchellh/mapstructure"
```

Comment:
Nit: Do we need mapstructure to decode the settings? Could we use std library from Go?

Reply:
We could cast the interface at each level to get to the `[]map[string]string` but it so easy with mapstructure and the code base and official doc from golangci mentioned using it (https://golangci-lint.run/contributing/new-linters/).